hack/test/unit-run.go

```diff
@@ -0,0 +1,220 @@
+	}
+	cmd := exec.Command("go", append(args, pkgs...)...)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	switch err := cmd.Run().(type) {
+	case nil, *exec.ExitError:
+		return cmd.ProcessState.ExitCode(), nil
+	default:
+		return 0, err
+	}
+}
+
+// mergeProfiles writes profile lines from each profile in profiles to final test profile at path.
+// mergeProfiles assumes that a valid profile file already exists at path.
+func mergeProfiles(path string, profiles ...*os.File) error {
+	// TODO: Consider using "golang.org/x/tools/cover" to parse profiles.
+	f, err := os.OpenFile(path, os.O_APPEND|os.O_WRONLY, 0)
+	if err != nil {
+		return fmt.Errorf("failed to open profile at %q", path)
+	}
+	for _, p := range profiles {
+		sc := bufio.NewScanner(p)
+		sc.Scan() // skip first line
+		for sc.Scan() {
+			if _, err := fmt.Fprintln(f, sc.Text()); err != nil {
+				return fmt.Errorf("failed to write profile line to %q: %w", path, err)
+			}
+		}
+		if err := sc.Err(); err != nil && err != io.EOF {
+			return fmt.Errorf("failed to scan %q: %w", p.Name(), err)
+		}
+	}
+	return nil
+}
+
+const (
+	TESTFLAGS = "TESTFLAGS"
+	TESTDIRS  = "TESTDIRS"
+	TIMEOUT   = "TIMEOUT"
+
+	profilePath = "bundles/profile.out"
+)
+
+func main() {
+	testDirBuf, err := exec.Command("go", "list", envOrDefault(TESTDIRS, "./...")).Output()
+	if err != nil {
+		log.Fatalf("Failed to list packages: %s", err)
+	}
+
+	testDirs := filterStrings(func(dir string) bool {
+		return !strings.HasPrefix(dir, pkgPath("integration")) // exclude integration tests
+	}, strings.Fields(string(testDirBuf))...)
+	if len(testDirs) == 0 {
+		log.Fatalf("No packages to test")
+	}
+
+	// NOTE: libnetwork tests require sequential execution.
+	// See https://github.com/moby/moby/issues/42458#issuecomment-873216754 for more info.
+
+	libnetwork, rest := partitionStrings(func(dir string) bool {
+		return strings.HasPrefix(dir, pkgPath("libnetwork"))
+	}, testDirs...)
+	if len(libnetwork) > 0 {
+		if func() bool { // true if "libnetwork/drivers/bridge" is going to be tested.
+			for _, dir := range libnetwork {
+				if dir == pkgPath("libnetwork/drivers/bridge") {
+					return true
+				}
+			}
+			return false
+		}() {
+			// Ensure docker-proxy is available and install if it is not.
+			if _, err := exec.LookPath("docker-proxy"); err != nil {
+				switch err := err.(type) {
+				case *exec.Error:
+					if err.Err != exec.ErrNotFound {
+						log.Fatalf("Failed to lookup `docker-proxy` executable in PATH: %s", err)
+					}
+				default:
+					log.Fatalf("Unknown error returned by `exec.LookPath`: %s", err)
+				}
+				if err := exec.Command("hack/make.sh", "binary-proxy", "install-proxy").Run(); err != nil {
+					log.Fatalf("Failed to install `docker-proxy`: %s", err)
+				}
+			}
+		}
+	}
+
+	var testFlags []string
+	if s, ok := os.LookupEnv(TESTFLAGS); ok {
+		testFlags = append(testFlags, strings.Fields(s)...) // TODO: Support flags containing spaces
```

Comment:
Is that OK for now? If not, maybe someone can suggest a simple way to handle this, which does not involve manually parsing complete flag string - doesn't seem like this is something we should be doing?